include/miopen/miopen.h

```diff
@@ -337,8 +337,24 @@ typedef enum
     miopenBFloat16 = 5, /*!< 16-bit binary floating point (8-bit exponent, 7-bit fraction)
                            (Partially supported) */
     miopenDouble = 6,   /*!< 64-bit floating point (Partially supported) */
+    miopenHalfx4 = 7,   /*!<Pack of 4 16-bit floating points in NCHW_VECT_C format*/
+    miopenHalfx8 = 8,   /*!<Pack of 8 16-bit floating points in NCHW_VECT_C format*/
 } miopenDataType_t;
 
+/*! @ingroup tensor
+ * @enum miopenTensorLayout_t
+ * MIOpen tensor layouts. NCHW, NHWC, CHWN and NCHW_VECT_C with certain vector length(vect_c=4,
+ * vect_c=8) are supported in MIOpen.
+ */
+typedef enum
+{
+    miopenTensorNCHW        = 0, /*!< NCHW memory layout (Fully supported) */
+    miopenTensorNHWC        = 1, /*!< NHWC memory layout (Fully supported) */
+    miopenTensorCHWN        = 2, /*!< CHWN memory layout (Not supported) */
+    miopenTensorNCHW_VECT_C = 3, /*!< NCHW_VECT_C memory layout (Partially supported) */
```

Comment:
API is not orthogonal, e.g. the same things can be expressed in different ways.
- For example, tensors of NCHW_VECT_C layout that contain non-vectorized datatype (e.g. HALF) are identical to conventional NCHW tensors.
- Another example, NHWC with non-packed datatype is identical to NHWC with packed datatype: NHWC HALF tensor is identical to both NHWC HALFx4 and NHWC HALFx8 tensors, provided that C is multiple of 4.

Shall we support this? I would prefer to support and explicitly document these specific cases, as this would allow more flexibility for the end user.

If we decide to not support these cases, then the restrictions must be explicitly documented.

Reply:
We do not support NCHW_VECT_C layout with non-vectorized datatype right now. I'll add the restrictions doc here.